keyvault/client/src/main/java/com/azure/keyvault/models/DeletedSecret.java

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.keyvault.models;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+public class DeletedSecret extends SecretAttributes {
+
+    /**
+     * The url of the recovery object, used to identify and recover the deleted
+     * secret.
+     */
+    @JsonProperty(value = "recoveryId")
+    private String recoveryId;
+
+    /**
+     * The time when the secret is scheduled to be purged, in UTC.
+     */
+    private OffsetDateTime scheduledPurgeDate;
+
+    /**
+     * The time when the secret was deleted, in UTC.
+     */
+    private OffsetDateTime deletedDate;
+
+    /**
+     * Get the recoveryId value.
+     *
+     * @return the recoveryId value
+     */
+    public String recoveryId() {
+        return this.recoveryId;
+    }
+
+    /**
+     * Set the recoveryId value.
+     *
+     * @param recoveryId the recoveryId value to set
+     * @return the DeletedSecret object itself.
+     */
+    public DeletedSecret recoveryId(String recoveryId) {
+        this.recoveryId = recoveryId;
+        return this;
+    }
```

Comment:
Should this class, and possibly others, be immutable?

Reply:
Deleted Secret can be made immutable.
But Secret and Secret Attributes cannot.

Internal classes for creating requests, do have a strong possibility of being immutable. Will double check and push the immutable changeset separately.